From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: splatage <noj.melville@gmail.com>
Date: Tue, 10 Mar 2026 22:00:00 +1300
Subject: [PATCH] Add Splatage config base


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 170d2bac2c280921248c86d266add46260c5a76d..7a67b9b43e33e54192be0a9df1f61103497bfb35 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -101,6 +101,7 @@ public class Main {

             }
 
             org.dreeam.leaf.config.LeafConfig.loadConfig(); // Leaf - Leaf config
+            io.splatage.leaf.config.SplatageConfig.loadConfig(); // Splatage - config
 
             CrashReport.preload();
             if (optionSet.has("jfrProfile")) { // CraftBukkit
@@ -119,6 +120,7 @@ public class Main {
             Bootstrap.validate();
             Util.startTimerHackThread();
             org.dreeam.leaf.config.ConfigModules.loadAfterBootstrap(); // Leaf - Leaf config - post load
+            io.splatage.leaf.config.SplatageConfigModules.loadAfterBootstrap(); // Splatage - config - post load
             Path path1 = Paths.get("server.properties");
             DedicatedServerSettings dedicatedServerSettings = new DedicatedServerSettings(optionSet); // CraftBukkit - CLI argument support
             dedicatedServerSettings.forceSave();
             
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 912ef41ae1e97ff856386394d8816710d31d2e13..afc72724132753b5438e9f753bc4f18c952f0d9c 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1352,6 +1352,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa

             // Paper end
             org.spigotmc.WatchdogThread.hasStarted = true; // Paper
             org.dreeam.leaf.config.LeafConfig.regSparkExtraConfig(); // Leaf - Leaf config
+            io.splatage.leaf.config.SplatageConfig.regSparkExtraConfig(); // Splatage - config
             // Paper start - Add onboarding message for initial server start
             if (io.papermc.paper.configuration.GlobalConfiguration.isFirstStart) {
                 LOGGER.info("*************************************************************************************");
